subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheConventionMappingIntegrationTest.groovy

```diff
@@ -37,8 +37,8 @@ class ConfigurationCacheConventionMappingIntegrationTest extends AbstractConfigu
                 id 'java-library'
             }
             tasks.withType(Jar.class).named('jar') { javaJar ->
-                // The Jar.archiveName field is of type Property<String>
-                javaJar.conventionMapping('archiveName') { 'some_name' }
+                // The Jar.archiveFileName field is of type Property<String>
+                javaJar.conventionMapping('archiveFileName') { 'some_name' }
```

Comment:
I don't think this is testing the same thing.

I think what was going on here is the Jar task had a `getArchiveName` and `setArchiveName`. This counts as a "archiveName property" in Groovy.  There was _also_ a field named `archiveName`. It looks like this is checking that CC does not try to set the Groovy archiveName property to the Java archiveName field.

I think to check this properly, we'd need the same scenario again.  Something like this:
```
abstract class MyTask extends ConventionTask {
    private final Property<String> archiveName 
    MyTask() {
      archiveName = project.objects.property(String)
    }
    void setArchiveName(String archiveName) { this.archiveName.set(archiveName) }
    void getArchiveName() { return archiveName.getOrNull() }

    @TaskAction 
    void doIt() {
        assert archiveName.get() == "something"
    }
}

task myTask(type: MyTask) {
    archiveName.convention("something")
    conventionMapping('archiveName') { 'not something' }
}
```

This is now emitting a deprecation warning around using a Provider type with convention mapping (that's not something we want to support). I asked on Slack why this wasn't failing the test.

Reply:
So I've been trying to get this to work, and I don't understand why `archiveName.convention("something")` shouldn't call the getter.

When I slightly alter the example to get it to run:

```
abstract class MyTask extends org.gradle.api.internal.ConventionTask {
  private final Property<String> archiveName  
  MyTask() {
    archiveName = project.objects.property(String)
  }
	  
  @Input String getArchiveName() { return archiveName.getOrNull() }
  void setArchiveName(String value) { this.archiveName.set(value) }
	  
  @TaskAction 
  void doIt() {
    assert archiveName.get() == "something"
  }
}

task myTask(type: MyTask) {
  archiveName.convention("something")
  conventionMapping('archiveName') { 'not something' }
}
```

I get:

```
A problem occurred evaluating root project 'test'.
> Cannot invoke method convention() on null object
```

on the line: `archiveName.convention("something")`.

It looks like the original code never tried to call `convention()` on the property itself, only `conventionMapping()`on the task.

Reply:
This works, but I don't think it tests what you wanted to test:

```
    def "doesn't restore convention value to incompatible field type"() {
        given:
        buildFile << """
            abstract class MyTask extends org.gradle.api.internal.ConventionTask {
                public final Property<String> archiveNameField
                MyTask() {
                    archiveNameField = project.objects.property(String)
                }
                
                @Input String getArchiveName() { return archiveNameField.getOrNull() }
                void setArchiveName(String value) { this.archiveNameField.set(value) }
            
                @TaskAction 
                void doIt() {
                    assert archiveNameField.get() == "something"
                }
            }
            
            task myTask(type: MyTask) {
                archiveNameField.convention("something")
                conventionMapping('archiveName') { 'not something' }
            }
        """

        expect: 'convention mapping is ignored'
        configurationCacheRun 'myTask'

        and: 'convention mapping is ignored just the same'
        configurationCacheRun 'myTask'
    }
```

Reply:
@big-guy Even after our conversation it seems I'm still not understanding what to do with this test.  I committed and pushed my last attempt to add a second getter that returns the field, leaving the field named the same as the getter/setter pair.  I understand this test is meant to check that the convention set on the property takes precedence over the conventionMapping directly on the field, but I don't understand how to make that happen in the test.